tests/docker/Dockerfile

```diff
@@ -64,6 +64,16 @@ RUN mkdir -p "/opt/kafka-2.3.1" && chmod a+rw /opt/kafka-2.3.1 && curl -s "$KAFK
     make install && \
     ldconfig
 
+# install dependencies for kafka clients such as sarama
+RUN git -C /tmp clone https://github.com/Shopify/sarama.git && \
+    cd /tmp/sarama/examples/interceptors && go mod tidy && go build && \
```

Comment:
you can't just do `go instal http://<tool>` ?

Reply:
On my local machine, I've experienced strange results with `go install` and `go get` where a few of the sarama examples are not downloaded. Thus, using git seemed like an OK alternative. I use each example in the compatibility testing but I did not try `go install` or `go get` in docker.

I'm trying those commands now in docker to see if I get the same result.

Reply:
It appears that installing from sarama via go get/install etc... doesn't install all of the binaries that we care about using the tests. so I think it makes sense to build in the dockerfile and install manually.

I'll also point out that we will have to revisit everything in the dockerfile in the near future when we decide start running ducktape on bare metal so we'll be able to revisit this (along with all the other dependencies).